src/ring_theory/polynomial/basic.lean

```diff
@@ -436,6 +436,75 @@ begin
   refl
 end
 
+lemma disjoint_ker_eval₂_of_coprime
+  (f : M →ₗ[R] M) {p q : polynomial R} (hpq : is_coprime p q) :
+  disjoint (p.eval₂ (algebra_map _ _) f).ker (q.eval₂ (algebra_map _ _) f).ker :=
```

Comment:
We can abbreviate `eval₂ (algebra_map _ _)` as `aeval`

Reply:
Oh, you've generalized it to support noncommutative rings. Awesome!